### PR TITLE
added methods

### PR DESCRIPTION
**Pull request by LizaBalshv**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -1,47 +1,144 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    static class Node<T> {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/MyLinkedList.java:7]: private
better put it at the end of the file

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -1,47 +1,144 @@
...
+        } else {
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index, size + 1);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/MyLinkedList.java:34]: image

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -1,47 +1,144 @@
...
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
```
Comment (okuzan) [inline src/main/java/core/basesyntax/MyLinkedList.java:63]: you can use for-each here

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -1,47 +1,144 @@
...
+        } else {
+            Node<T> prev = getNodeByIndex(index - 1);
+            removedValue = prev.next.value;
+            prev.next = prev.next.next;
+            if (index == size - 1) {
+                last = prev;
+            }
+        }
+        size--;
+        return removedValue;
```
Comment (okuzan) [inline src/main/java/core/basesyntax/MyLinkedList.java:105]: create unlink method for `remove`s

**Review by okuzan — CHANGES_REQUESTED** [review]
Make sure to read common mistakes [checklist](https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md) attached to the task description before creating a PR 🙏

**Review by olekskov — APPROVED** [review]

**Pull request closed without merging**